### PR TITLE
feat: update Dockerfile to deploy

### PR DESCRIPTION
**Pull request by xiongsp**
If this PR is related to coding or code translation, please fill out the checklist.

This PR is not related to coding or code translation. This PR solves #134.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM python:3.9.0-buster
+
+RUN sed -i 's/deb.debian.org/mirrors.aliyun.com/g' /etc/apt/sources.list
+RUN sed -i 's/security.debian.org/mirrors.aliyun.com/g' /etc/apt/sources.list
+
+RUN pip install --upgrade pip && pip install mkdocs mkdocs-material
+
+WORKDIR /app
+
+COPY codes /app/codes
```
Comment (krahets) [inline Dockerfile:10]: Maybe `Add . /app` is clearer.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+FROM python:3.9.0-buster
+
+RUN sed -i 's/deb.debian.org/mirrors.aliyun.com/g' /etc/apt/sources.list
```
Comment (krahets) [inline Dockerfile:3]: Is this setting only for Linux? How about Win and macOS?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+RUN sed -i 's/security.debian.org/mirrors.aliyun.com/g' /etc/apt/sources.list
+
+RUN pip install --upgrade pip && pip install mkdocs mkdocs-material
+
+WORKDIR /app
```
Comment (krahets) [inline Dockerfile:8]: Could we use `hello-algo` instead of `app`?

**Review by krahets — COMMENTED** [review]
Thank you for adding the dockerfile. Please address the comments.

File: Dockerfile (hunk 1 above)
Comment (xiongsp) [reply Dockerfile:10]: If use `COPY . /app`, other files like `LICENSE` or `.gitignore` will be copied into the **DOCKER**, which is unnecessary.

File: Dockerfile (hunk 2 above)
Comment (xiongsp) [reply Dockerfile:3]: NO. `RUN` means **to run in the docker** but not the **host**. So there is no compatible problem. But here I just made a mistake, I will correct it later.

File: Dockerfile (hunk 3 above)
Comment (xiongsp) [reply Dockerfile:8]: YES, we can. But it is unnecessary for it is stored inside the **docker**. Nobody can see its name unless use `docker exec -it xxx` to get a shell in the docker. `app` is the tradition name for python service.

**Comment by xiongsp** [comment]
If necessary, how about adding instruction in `README` or somewhere else guiding users to deploy the docker?

File: Dockerfile (hunk 2 above)
Comment (xiongsp) [reply Dockerfile]: Now it works correctly.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,17 @@
+FROM python:3.9.0-alpine
```
Comment (xiongsp) [inline Dockerfile:1]: This aims to reduce the size of images from ~1G to ~180M

File: README.md [hunk 5]
```diff
@@ -41,6 +41,28 @@
 
 如果感觉本书对你有所帮助，请点个 Star :star: 支持一下，谢谢！
 
+### 本地部署
```
Comment (krahets) [inline README.md:44]: Please move the tutorial to `docs/chapter_preface/contribution.md`. Docker is not for most readers.

**Review by krahets — COMMENTED** [review]
Thanks, @xiongsp! Please address the comment. I've modified the Dockerfile. Could you help test it?

**Comment by xiongsp** [comment]
@krahets, I noticed that you split up the RUN command. Although it would increase the size of images, it works and the increasement of the size is bearable. The Dockerfile has been tested and it works correctly. And the tutorial has been moved to `docs`.

File: Dockerfile (hunk 4 above)
Comment (krahets) [reply Dockerfile:1]: Great change.

**Review by krahets — APPROVED** [review]
Thanks for the great PR!

**Comment by krahets** [comment]
> @krahets, I noticed that you split up the RUN command. Although it would increase the size of images, it works and the increasement of the size is bearable. The Dockerfile has been tested and it works correctly. And the tutorial has been moved to `docs`.

I see. I'll merge them.

**Comment by krahets** [comment]
> @krahets, I noticed that you split up the RUN command. Although it would increase the size of images, it works and the increasement of the size is bearable. The Dockerfile has been tested and it works correctly. And the tutorial has been moved to `docs`.

I see. I'll merge them.

**Pull request merged**